math/src/main/scala/breeze/linalg/support/CanTraverseValues.scala

```diff
@@ -52,81 +65,105 @@ trait CanTraverseValues[From, A] {
 
 object CanTraverseValues {
 
-  trait ValuesVisitor[@specialized A] {
-    def visit(a: A)
-    def visitArray(arr: Array[A]):Unit = visitArray(arr, 0, arr.length, 1)
+  /**
+    * This trait can be implemented and used to apply a function over a breeze collection/tensor,
+    * using [[CanTraverseValues.traverse()]].
+    *
+    * @tparam V
+    */
+  trait ValuesVisitor[@specialized V] {
 
-    def visitArray(arr: Array[A], offset: Int, length: Int, stride: Int):Unit = {
+    def visit(a: V)
+
+    /** Visits an array assuming offset of zero and stride of 1 */
+    final def visitArray(arr: Array[V]): Unit = visitArray(arr, 0, arr.length, 1)
+
+    def visitArray(arr: Array[V], offset: Int, length: Int, stride: Int): Unit = {
       import spire.syntax.cfor._
       // Standard array bounds check stuff
       if (stride == 1) {
-        cforRange(offset until length + offset) { i =>
-          visit(arr(i))
-        }
+        cforRange(offset until length + offset) { i => visit(arr(i)) }
       } else {
-        cforRange(0 until length) { i =>
-          visit(arr(i * stride + offset))
-        }
+        cforRange(0 until length) { i => visit(arr(i * stride + offset)) }
       }
     }
-    def zeros(numZero: Int, zeroValue: A)
+
+    def zeros(numZero: Int, zeroValue: V): Unit
+
```

Comment:
I see.

Are you happy with the ```zeros``` naming? It sounds a bit to me like a dense array constructor filled with zeros?  How about ```visitZeros```?